### PR TITLE
Model

### PR DESCRIPTION
**Pull request by KirillDogadin-std**
Hope i understood the conversion logic correctly. 

There a lot of tuples returned in the methods I templated because as I understood the return value should go to the loss function. And since those loss functions get e.g. 3 arguments, i had to return 3 arguments. In case that assumption was wrong, please correct me.

### CODE REVIEW

File: code/pytorch/MultiKE_model.py [hunk 1]
```diff
@@ -79,45 +79,61 @@ def relation_view(self, rel_pos_hs, rel_pos_rs, rel_pos_ts, rel_neg_hs, rel_neg_
...
+        rel_nhs = torch.index_select(rv_ent_embeds, dim=0, index=rel_neg_hs)
+        rel_nrs = torch.index_select(rel_embeds, dim=0, index=rel_neg_rs)
+        rel_nts = torch.index_select(rv_ent_embeds, dim=0, index=rel_neg_ts)
         return rel_phs, rel_prs, rel_pts, rel_nhs, rel_nrs, rel_nts
 
     def attribute_view(self, attr_pos_hs, attr_pos_as, attr_pos_vs):
-        attr_phs = tf.nn.embedding_lookup(self.av_ent_embeds, attr_pos_hs)
-        attr_pas = tf.nn.embedding_lookup(self.attr_embeds, attr_pos_as)
-        attr_pvs = tf.nn.embedding_lookup(self.literal_embeds, attr_pos_vs)
+        # Question: Why not l2_normalize? like in `relation_view`
```
Comment (KirillDogadin-std) [inline code/pytorch/MultiKE_model.py:90]: Question in this comment

File: code/pytorch/MultiKE_model.py [hunk 2]
```diff
@@ -79,53 +80,111 @@ def relation_view(self, rel_pos_hs, rel_pos_rs, rel_pos_ts, rel_neg_hs, rel_neg_
...
+        else:  # wavg
+            ent_embeds = model.ent_embeds
+        print(embed_choice, 'test results:')
+        embeds1 = ent_embeds[model.kgs.test_entities1, ]
+        embeds2 = ent_embeds[model.kgs.test_entities2, ]
+        hits1_12, mrr_12 = eva.valid(embeds1, embeds2, None, model.args.top_k, model.args.test_threads_num,
+                                     normalize=True)
+        del embeds1, embeds2
+        gc.collect()
+        return mrr_12
```
Comment (KirillDogadin-std) [inline code/pytorch/MultiKE_model.py:187]: not sure if the approach was correct here.

File: code/pytorch/MultiKE_model.py [hunk 3]
```diff
@@ -79,53 +80,111 @@ def relation_view(self, rel_pos_hs, rel_pos_rs, rel_pos_ts, rel_neg_hs, rel_neg_
...
+        else:  # 'final'
+            ent_embeds = model.ent_embeds
+        print(embed_choice, 'valid results:')
+        embeds1 = ent_embeds[model.kgs.valid_entities1, ]
+        embeds2 = ent_embeds[model.kgs.valid_entities2 + model.kgs.test_entities2, ]
+        hits1_12, mrr_12 = eva.valid(embeds1, embeds2, None, model.args.top_k, model.args.test_threads_num,
+                                     normalize=True)
+        del embeds1, embeds2
+        gc.collect()
+        return mrr_12
```
Comment (KirillDogadin-std) [inline code/pytorch/MultiKE_model.py:162]: as above
